app/controllers/proposals_controller.rb

```diff
@@ -50,7 +50,7 @@ def cancel
   end
 
   def approve
-    approval = proposal.existing_approval_for(current_user)
+    approval = proposal.existing_step_for(current_user)
```

Comment:
ah! good point. The word "approve" and derivatives are overloaded in this app.

I guess the assumption is that `current_user` would only ever have one pending step of any given flavor at a time?